### PR TITLE
Remove bind_email parameter from register_user

### PR DESCRIPTION
**Pull request by anoadragon453**
As far as I can tell, this was only used by the `bind_email` parameter which was removed in https://github.com/matrix-org/synapse/pull/5964.

Emails added during registration still happen in `post_registration_actions`:

https://github.com/matrix-org/synapse/blob/2d6d4fa8aff7187b33fa00147703a802c1c80c95/synapse/handlers/register.py#L556-L565

### CODE REVIEW

File: synapse/handlers/register.py [hunk 1]
```diff
@@ -140,7 +140,6 @@ def register_user(
         user_type=None,
         default_display_name=None,
         address=None,
-        bind_emails=[],
```
Comment (clokep) [inline synapse/handlers/register.py:143]: Please remove this from the docstring as well.

File: synapse/module_api/__init__.py [hunk 2]
```diff
@@ -89,7 +89,7 @@ def check_user_exists(self, user_id):
         return defer.ensureDeferred(self._auth_handler.check_user_exists(user_id))
 
     @defer.inlineCallbacks
-    def register(self, localpart, displayname=None, emails=[]):
+    def register(self, localpart, displayname=None):
```
Comment (clokep) [inline synapse/module_api/__init__.py:92]: We should probably note in the changelog that his changed (or leave it for backwards compatibility).

Do we know if any external modules called this with an iterable of emails?

Comment (anoadragon453) [reply synapse/module_api/__init__.py:92]: Aha! Good spot. Ok, I was under the impression that modules like matrix-synapse-ldap3 bound emails in another manner (through the `check_3pid_auth` module method).

But [you're right](https://github.com/matrix-org/matrix-synapse-ldap3/blob/490b2bf844b3dab2933fb92029714fbeb9e086b0/ldap_auth_provider.py#L294-L296)!

Which probably makes this whole PR defunct I suppose.

Comment (clokep) [reply synapse/module_api/__init__.py:92]: Ah, yeah looks like this is still used then. :)

Comment (anoadragon453) [reply synapse/module_api/__init__.py:92]: Oh well!

**Comment by anoadragon453** [comment]
Closing as this argument is still used by modules. Your editor may not show it though as it won't know about a separate codebase's method calls :)

**Pull request closed without merging**